shell/platform/darwin/common/framework/Source/FlutterNSBundleUtils.mm

```diff
@@ -61,10 +61,23 @@
   NSString* flutterAssetsPath = FLTAssetPath(bundle);
   // Use the raw path solution so that asset path can be returned from unloaded bundles.
   // See https://github.com/flutter/engine/pull/46073
-  NSString* assetsPath = [bundle pathForResource:flutterAssetsPath ofType:@""];
+  NSString* assetsPath = [bundle pathForResource:flutterAssetsPath ofType:nil];
+  if (assetsPath.length == 0) {
+    // In app extension, using full relative path(kDefaultAssetPath)
+    // returns nil when the app bundle is not loaded. We try to use
+    // the sub folder name, which can successfully return a valid path.
+    assetsPath = [bundle pathForResource:@"flutter_assets" ofType:nil];
+  }
+
+  if (assetsPath.length == 0) {
+    assetsPath = [[NSBundle mainBundle] pathForResource:flutterAssetsPath ofType:nil];
+  }
 
   if (assetsPath.length == 0) {
-    assetsPath = [[NSBundle mainBundle] pathForResource:flutterAssetsPath ofType:@""];
+    // In app extension, using full relative path(kDefaultAssetPath)
```

Comment:
Same comments here. But rather than duplicating everything including the comments twice, let's extract a private (`static`) helper function which tries only the given bundle, then this function can call the private one twice, once passing the passed bundle and once passing the main bundle.